packages/flutter_tools/lib/src/flutter_cache.dart

```diff
@@ -235,18 +235,19 @@ class FlutterSdk extends EngineCachedArtifact {
 
   @override
   List<List<String>> getBinaryDirs() {
-    // Currently only Linux supports both arm64 and x64.
+    // Linux and Windows both support arm64 and x64.
     final String arch = cache.getHostPlatformArchName();
     return <List<String>>[
       <String>['common', 'flutter_patched_sdk.zip'],
       <String>['common', 'flutter_patched_sdk_product.zip'],
       if (cache.includeAllPlatforms) ...<List<String>>[
-        <String>['windows-x64', 'windows-x64/artifacts.zip'],
+        <String>['windows-$arch', 'windows-$arch/artifacts.zip'],
         <String>['linux-$arch', 'linux-$arch/artifacts.zip'],
         <String>['darwin-x64', 'darwin-$arch/artifacts.zip'],
       ]
-      else if (_platform.isWindows)
-        <String>['windows-x64', 'windows-x64/artifacts.zip']
+      else if (_platform.isWindows) ...<List<String>>[
```

Comment:
nit. after the refactor, this can be simplified to be just a single `List<String>` literal, we don't need the spread operator  and nested list